.travis.yml

```diff
@@ -31,14 +29,25 @@ script:
   - npm test
   - python manage.py test --with-coverage
   - py.test --cov-append --cov-report=term --cov=. -v
-  # This temporarily exits 0 while we cleanup
   - ./node_modules/.bin/eslint .
   - codecov
 
   # Heroku deployment file is valid
   - cat app.json | python -m json.tool > /dev/null
+
 addons:
   postgresql: "9.4"
+  apt:
+    packages:
+      - language-pack-tr
+
+before_cache:
+  # workaround for travis/aufs to prevent cache upload if no change
+  - pip uninstall py pytest -y
+  - pip install py==1.4.34 pytest==3.2.3
```

Comment:
This looks bad. First it's setting the version number in a different place (so we'll have to remember to update it here when we do in `requirements.txt`) and it doesn't require hashes, opening some security issues. Wouldn't it be better to maybe extract those two deps from the `requirements.txt` file in order to be able to install just those two, and still have only one place where we set the version and hash?

Reply:
Yep, the hardcoding requirements is a tradeoff - but imhe travis doesnt change these very often, probs ~every 6 months or so.

> Wouldn't it be better to maybe extract those two deps from the requirements.txt

Its not about our requirements - its about travis provided versions - i can test this further but i dont think this is optional if you want to save ~30s on every test run

Reply:
Huh, travis uses its own version of pytest, which is not necessarily the one we have decided upon in our `requirements.txt`???

Reply:
it uses ours, but for this hacky kludgy, but very effective, workaround to work we need to reset to travis' version at the end of each test run